### DISCUSSION TITLE
Managing asset traits

### DISCUSSION THREAD

**Discussion by hanseartic** [stellar/sorobanathon#160] [General]
I am working on an implementation of a trait contract as follows:

> # 📜 A Trait Contract
>
>Collections (of NFTs) often are associated with a set/combination of traits pulled from a pool of traits. This soroban contract aims at allowing to store the initial pool of traits and then generating a random combination of trait-options from the remaining pool until all traits / options have been drawn from the pool.

Find the work here: https://github.com/hanseartic/sorobanathon/pull/1

**Comment by achallagundla** [comment]
Heya,
Congratulations! Your contribution to Sorobanathon: First Light has been accepted! 🎉
Your submission may qualify for a reward of 1100 XLM! One last step: to receive that reward, please fill out this [form](https://stellarform.typeform.com/Sorobanathon) completely and accurately.

Please let us know if you have any questions. (You can also message me on discord @ anuxhya)

All the best,
Anuhya

PS: No need to submit the form if you already have, just update me on discord if your wallet address has changed.
